pages/album/[id].js

```diff
@@ -1,30 +1,56 @@
 import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
 import AlbumDetail from "../../components/AlbumDetail/AlbumDetail.js";
-import curtisAlbum from "../../public/images/curtis.jpeg";
 import AppHeader from "../../components/Header/Header.js";
 
 export default function AlbumDetailPage() {
   const router = useRouter();
-  const { id } = router.query;
+  const { id, coverImage } = router.query;
+  const [album, setAlbum] = useState(null);
 
-  const album = {
-    name: "Curtis Mayfield",
-    title: "Curtis (Deluxe Edition)",
-    genre: "Funk, Soul",
-    year: 1970,
-    id: id,
-    src: curtisAlbum,
-  };
+  useEffect(() => {
+    async function fetchAlbum() {
+      try {
+        const response = await fetch(`https://api.discogs.com/releases/${id}`);
+        const data = await response.json();
+
+        if (response.ok) {
+          const albumData = {
+            id: data.id,
+            name: data.artists[0].name,
+            title: data.title,
+            genre: data.genres[0],
+            year: data.year,
+            src: coverImage || data.images[0].resource_url, // use coverImage if available, otherwise use first image from data
+          };
+
+          setAlbum(albumData);
+        } else {
+          console.error("Failed to fetch album data:", data.message);
+        }
+      } catch (error) {
+        console.error("Error fetching album data:", error);
+      }
+    }
+
+    if (id) {
+      fetchAlbum();
+    }
+  }, [id, coverImage]);
 
   return (
     <>
       <AppHeader />
-      <AlbumDetail album={album}>
-        <p>{album.name}</p>
-        <p>{album.title}</p>
-        <p>{album.genre}</p>
-        <p>{album.year}</p>
-      </AlbumDetail>
+      {album ? (
+        <AlbumDetail album={album}>
+          <p>{album.name}</p>
+          <p>{album.title}</p>
+          <p>{album.genre}</p>
+          <p>{album.year}</p>
+        </AlbumDetail>
```

Comment:
It's a bit strange to pass this down as `children`. It feels like the AlbumDetail component should lay out these information on it's own.

```suggestion
        <AlbumDetail album={album} />
```

Reply:
You can move…

```js
          <p>{album.name}</p>
          <p>{album.title}</p>
          <p>{album.genre}</p>
          <p>{album.year}</p>
```

…into the AlbumDetail component